scholarly_articles/scripts/set_country.py

```diff
@@ -2,20 +2,27 @@
 
 from scholarly_articles import models
 from usefulmodels.models import Country
+from institution.models import Institution
 
 
 def run():
-    #first iteration to identify country by name
+    #first iteration to identify country by institution ROR
+    for institution_ror in Institution.objects.filter(source="ROR").iterator():
+        for aff in models.Affiliations.objects.filter(name__icontains=institution_ror.name,
+                                                      country__isnull=True).iterator():
+            aff.country = institution_ror.location.country
+
+    #second iteration to identify country by declared name
     for country in Country.objects.all():
         for aff in models.Affiliations.objects.filter(
                 Q(name__icontains=country.name_en) | Q(name__icontains=country.name_pt),
                   country__isnull=True, official__isnull=True).iterator():
             aff.country = country
             aff.save()
 
-        #second iteration to identify country by acronym with 3 char
+        #second iteration to identify country by declared acronym with 3 char
         for aff in models.Affiliations.objects.filter(
-                name__icontains=country.acron3,
+                Q(name__contains=" "+country.acron3+" ") | Q(name__contains=" "+country.acron3),
```

Comment:
@Rossi-Luciano Talvez melhor restringir para `Q(name__contains=", " + country.acron3+", ") | Q(name__endswith=", "+country.acron3)`, ou seja, procurar por um padrão `, ACRON3,` ou  que termine em `, ACRON3`. Aí pode-se fazer uma nova rodada com for para `ACRON2` buscando pelo mesmo padrão, se avaliar que vale a pena.